org.eclipse.xtext.builder/src/org/eclipse/xtext/builder/JDTAwareEclipseResourceFileSystemAccess2.java

```diff
@@ -63,12 +63,14 @@ protected void createContainer(IContainer container) throws CoreException {
 	protected void addToSourceFolders(IContainer container) throws JavaModelException {
 		IJavaProject jp = JavaCore.create(container.getProject());
 		if (jp.exists() && !jp.isOnClasspath(container)) {
-			IPackageFragmentRoot currentSource = jp.findPackageFragmentRoot(jp.getPath().append(getCurrentSource()));
-			if (currentSource != null) {
-				IClasspathEntry currentClasspathEntry = currentSource.getRawClasspathEntry();
-				if (currentClasspathEntry != null) {
-					insertClasspathEntry(container, currentClasspathEntry, jp);
-					return;
+			if (getCurrentSource() != null) {
```

Comment:
hmmm i just kept it consistent with other places. but i can refactor that as well